# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/net-snmp/scapi.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -Naur ucd-snmp-4.2.6-/snmplib/scapi.c ucd-snmp-4.2.6/snmplib/scapi.c
--- ucd-snmp-4.2.6-/snmplib/scapi.c	2001-10-01 10:52:49.000000000 +0100
+++ ucd-snmp-4.2.6/snmplib/scapi.c	2003-04-08 15:13:09.000000000 +0100
@@ -609,7 +609,7 @@
 	memset(my_iv, 0, sizeof(my_iv));
 	memset(pad_block, 0, sizeof(pad_block));
 	memset(key_struct, 0, sizeof(key_struct));
-	memset(key_sch, 0, sizeof(key_sch));
+	memset((char*)&key_sch, 0, sizeof(key_sch));
 	return rval;
 
 }  /* end sc_encrypt() */
@@ -722,7 +722,7 @@
 
 /* exit cond */
 sc_decrypt_quit:
-	memset(key_sch, 0, sizeof(key_sch));
+	memset((char*)&key_sch, 0, sizeof(key_sch));
 	memset(key_struct, 0, sizeof(key_struct));
 	memset(my_iv, 0, sizeof(my_iv));
 	return rval;
